tests/model_registry/image_validation/test_verify_rhoai_images.py

```diff
@@ -0,0 +1,88 @@
+import pytest
+from typing import Self
+from simple_logger.logger import get_logger
+from kubernetes.dynamic import DynamicClient
+from pytest_testconfig import config as py_config
+
+from utilities.constants import DscComponents, Labels
+from utilities.general import (
+    get_csv_related_images,
+    get_pods_by_labels,
+    validate_container_images,
+)
+from tests.model_registry.constants import MR_OPERATOR_NAME, MR_INSTANCE_NAME, MR_NAMESPACE
+from ocp_resources.model_registry import ModelRegistry
+
+LOGGER = get_logger(name=__name__)
+
+
+@pytest.mark.parametrize(
+    "updated_dsc_component_state_scope_class",
+    [
+        {
+            "component_patch": {
+                DscComponents.MODELREGISTRY: {
+                    "managementState": DscComponents.ManagementState.MANAGED,
+                    "registriesNamespace": MR_NAMESPACE,
+                }
+            }
+        }
+    ],
+    indirect=True,
+)
+@pytest.mark.usefixtures("updated_dsc_component_state_scope_class")
+class TestModelRegistryImages:
+    """
+    Tests to verify that all Model Registry component images (operator and instance container images)
+    meet the requirements:
+    1. Images are hosted in registry.redhat.io
+    2. Images use sha256 digest instead of tags
+    3. Images are listed in the CSV's relatedImages section
+    """
+
+    @pytest.mark.smoke
+    def test_verify_model_registry_images(
+        self: Self, admin_client: DynamicClient, model_registry_instance: ModelRegistry
+    ):
+        # Get related images from CSV
+        related_images = get_csv_related_images(admin_client=admin_client)
+        related_image_refs = {img["image"] for img in related_images}
+
+        # Get operator pod
+        operator_pod_list = get_pods_by_labels(
+            admin_client=admin_client,
+            namespace=py_config["applications_namespace"],
+            label_selector=f"{Labels.OpenDataHubIo.NAME}={MR_OPERATOR_NAME}",
+        )
+        if len(operator_pod_list) > 1:
+            LOGGER.warning(f"Expected 1 operator pod, found {len(operator_pod_list)}. Using first one.")
+        operator_pod = operator_pod_list[0]
+
+        # Get instance pod
+        instance_pod_list = get_pods_by_labels(
+            admin_client=admin_client,
+            namespace=MR_NAMESPACE,
+            label_selector=f"app={MR_INSTANCE_NAME}",
+        )
+        if len(instance_pod_list) > 1:
+            LOGGER.warning(f"Expected 1 instance pod, found {len(instance_pod_list)}. Using first one.")
+        instance_pod = instance_pod_list[0]
```

Comment:
We have very similar codes here. Can we have utils that gets the pods and validates their images?

Reply:
IMHO this is good as is, the getting of the pods and image validation are already defined as external utils, the code in the test takes care of test-specific steps, i.e. how many pods I expect to find with my label selector

Reply:
This can be simplified. As you can see you have same lines of code that calls get_pods_by_labels() and checks the length. Also, if we don't expect >1 pods, should not we fail?

Reply:
What do you mean by simplified? This code still needs to live somewhere. It is highly unlikely that these same steps will be needed elsewhere other than this test, so to me it makes the most sense to have the code here rather than abstracted in yet another function that only gets called from here

Reply:
Let's add a wait then.

Reply:
done